templates.js

```diff
@@ -799,5 +799,382 @@ window.addEventListener('message', function(event){
+
+    `,
+    arguments: [ 
+    { id: 'products',
+      type: 'text',
+      displayName: 'Products',
+      recommended: "products",
+      description: 'Variable name that stores the products in your virtual shopping cart',
+      default: "[remove] products"
+    },  
+    { id: 'productSku',
+      type: 'text',
+      displayName: 'Product SKU',
+      recommended: "sku",
+      description: 'Product stock-keeping unit (required)',
+      default: "product.id"
+    }, 
+    { id: 'productName',
+    type: 'text',
+    displayName: 'Product name',
+    recommended: "name",
+    description: 'The name of the product (optional)',
+    default: "product.name"
+    }, 
+    { id: 'productCategory',
+    type: 'text',
+    displayName: 'Product category',
+    recommended: "category",
+    description: 'Product category (optional)',
+    default: "product.category"
+    }, 
+    { id: 'productPrice',
+    type: 'text',
+    displayName: 'Product price',
+    recommended: "price",
+    description: 'The price of the product (optional)',
+    default: "product.price"
+    }, 
+    { id: 'productQuantity',
+    type: 'text',
+    displayName: 'Product quantity',
+    recommended: "quantity",
+    description: 'The number of units (optional)',
+    default: "product.quantity"
+    },   
+    { id: 'productBrand',
+    type: 'text',
+    displayName: 'Product brand',
+    recommended: "brand",
+    description: 'The brand of the product (optional)',
+    default: "product.brand"
+    },  
+    { id: 'productVariant',
+    type: 'text',
+    displayName: 'Product variant',
+    recommended: "variant",
+    description: 'The variant of the product (optional)',
+    default: "product.variant"
+    },      
+   ]
+  },
+  {
+    id: 'EcommerceOrder',
+    name: 'Ecommerce order',
+    description: `
+    This script lets you track an ecommerce order. You can add it as a tag with a data layer event trigger in Tag Manager. You can see collected ecommerce data in Analytics > Ecommerce or use them in Analytics > Custom reports. Note:  if you have an enchanced ecommerce setup, [purchase] products should be a dataLayer variable with the value taken from ecommerce.purchase.products and actionField should be a dataLayer varialbe with the value taken from ecommerce.purchase.actionField
+    `,
+    template: `
+
+  window._paq = window._paq || [];
+  var products = {{ {{products}} }}.map(function(product) {
+    return {
+      sku: {{productSku}},
+      name: {{productName}},
+      category: {{productCategory}},
+      price: {{productPrice}},
+      brand: {{productBrand}},
+      variant: {{productVariant}},
+      quantity: {{productQuantity}}
+    };
+  });
+  
+  var paymentInformation = [{{ {{actionField}} }}].map(function(order) {
+    return {
+      orderId: {{orderId}},
+      grandTotal: {{grandTotal}},
+      subTotal: {{subTotal}},
+      tax: {{tax}},
+      shipping: {{shipping}}
+    };
+  });
+  paymentInformation = paymentInformation[0];
```

Comment:
Why only first item? do we create [array in here](https://github.com/PiwikPRO/solutions-templates/pull/46/files#diff-2a2150d0b57522a5f463ffea155f57cc31bf3439fc4953b00b0bca842e89970bR1064) to just map over it and not use as typical array? Hw about to remove that [] + [0] then?

Reply:
Yes, I tried to solve it similarly to the above since actionField is just an object and not an array (map function only works on arrays), I've placed it as an object inside of array and took out the first index (since there will always be 1). I was also thinking about going with the answer provided here - https://stackoverflow.com/questions/14810506/map-function-for-objects-instead-of-arrays